### ISSUE TITLE
Equality alignment with guards

### ISSUE THREAD

**Issue by marcosh** [commercialhaskell/hindent#527]
## Question

If I have a function with a guard like

```
f :: [Int] -> Int
f (x : xs) | x' > 5 = 37
f _ = 13
``` 

I would expect `hindent` to align the `= signs`, as in

```
f :: [Int] -> Int
f (x : xs) | x' > 5 = 37
f _                 = 13
```
Instead it aligns the `=` of the second line with the `|` of the first, as in

```
f :: [Int] -> Int
f (x : xs) | x' > 5 = 37
f _        = 13
```

Is this the desired and correct behaviour?

**Comment by toku-sa-n** [comment]
Actually, [`stylish-haskell`](https://github.com/haskell/stylish-haskell) does this kind of job, although it doesn't format your input well as you're using a guard. It formats the following input

```haskell
f :: [Int] -> Int
f (x : xs) = 37
f _ = 13
```

to

```haskell
f :: [Int] -> Int
f (x : xs) = 37
f _        = 13
```

@mihaimaruseac Do we close this issue as it's `stylish-haskell`'s job, or keep it open as a feature request?

**Comment by toku-sa-n** [comment]
Agreed. After reviewing other issues, I've decided to keep this one open as there is another align-related issue (#230). It may be good to let HIndent consider alignments in the future.

**Comment by toku-sa-n** [comment]
Maybe we can implement this by using [`format`](https://hackage.haskell.org/package/stylish-haskell-0.14.4.0/docs/Language-Haskell-Stylish.html#v:format) defined in [`stylish-haskell`](https://hackage.haskell.org/package/stylish-haskell-0.14.4.0).
